src/core/namespace.c

```diff
@@ -56,10 +56,10 @@ typedef enum MountMode {
         PRIVATE_TMP,
         PRIVATE_VAR_TMP,
         PRIVATE_DEV,
+        READONLY,
         BIND_DEV,
         SYSFS,
         PROCFS,
-        READONLY,
```

Comment:
this appears unnecessary. Note that we actually propagate read-only flags to the remaining MountEntry if we drop one because we have a duplicate definition. This means, if you use ReadOnlyPaths=/proc in combination with MountAPIVS=true, then you should get a MountEntry as result that has mode=BIND_DEV but also read_only=true.

This patch looks incorrect to me. Did you verify that something is actually broken here?